tests/ui/consts/static-mut-refs.rs

```diff
@@ -17,7 +17,9 @@ static mut INT_RAW: *mut isize = &mut 1isize as *mut _;
 pub fn main() {
     unsafe {
         TEST[0] += 1;
+        //~^ WARN creating a reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.